python/pyspark/sql/types.py

```diff
@@ -505,6 +508,9 @@ def __eq__(self, other):
 
 def _parse_datatype_json_string(json_string):
     """Parses the given data type JSON string.
+    >>> import pickle
```

Comment:
It looks like this is a regression test for the singleton `__eq__` issue.  It feels a little odd to have this as a doctest as opposed to a dedicated `unittest` regression test.  Maybe we could move this into a new test case and add a comment referencing this JIRA?